misaligned.cpp

```diff
@@ -1,19 +1,53 @@
 #include <assert.h>
 #include <iostream>
+#include <map>
+#include <string>
+enum MajorColor{ White = 0, Red, Black, Yellow, Violet };
+enum MinorColor{ Blue = 0, Orange, Green, Brown, Slate };
+const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+int numberOfMajorColors = sizeof(majorColor) / sizeof(majorColor[0]);
+int numberOfMinorColors = sizeof(minorColor) / sizeof(minorColor[0]);
+
+std::map <std::string , MajorColor> majorColorEnumtoStringMap;
+std::map <std::string , MinorColor> minorColorEnumtoStringMap;
+
+void mapMajorarraytoEnum() {
+    for (int i = 0; i < numberOfMajorColors; i++) {
+        majorColorEnumtoStringMap.insert({ majorColor[i] , (MajorColor)i });
+    }
+}
+
+void mapMinorarraytoEnum() {
+    for (int i = 0; i < numberOfMinorColors; i++) {
+        minorColorEnumtoStringMap.insert({ minorColor[i] , (MinorColor)i });
+    }
+}
+
+int mapColorToNumber(MajorColor majorColor, MinorColor minorColor) {
+    int pairNumber = majorColor * 5 + minorColor;
+    return pairNumber;
+}
 
 int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
     for (i = 0; i < 5; i++) {
         for (j = 0; j < 5; j++) {
             std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            int pairNumber = mapColorToNumber(majorColorEnumtoStringMap[majorColor[i]] ,
+                                                minorColorEnumtoStringMap[minorColor[i]]);
+            assert(pairNumber == (i * 5 + j));

```

Comment:
This `assert` is used to check the goodness of the logic. Once proven, it should never fail, unless we change the code again.
Hence it is enough to have the assert in test code to check the code before releasing. It's not necessary to execute it every time in production. Separate the assert into test code.